rasa/nlu/extractors/extractor.py

```diff
@@ -148,11 +149,32 @@ def convert_predictions_into_entities(
             current_group_tag = self.get_tag_for(tags, ENTITY_ATTRIBUTE_GROUP, idx)
             current_role_tag = self.get_tag_for(tags, ENTITY_ATTRIBUTE_ROLE, idx)
 
-            new_tag_found = (
-                last_entity_tag != current_entity_tag
-                or last_group_tag != current_group_tag
-                or last_role_tag != current_role_tag
-            )
+            if bilou_utils.bilou_prefix_from_tag(current_entity_tag):
+                new_tag_found = (
+                    last_entity_tag == NO_ENTITY_TAG
+                    or (
+                        last_entity_tag != current_entity_tag
+                        and (
+                            bilou_utils.LAST
+                            != bilou_utils.bilou_prefix_from_tag(current_entity_tag)
+                            and bilou_utils.INSIDE
+                            != bilou_utils.bilou_prefix_from_tag(current_entity_tag)
+                        )
+                    )
+                    or last_group_tag != current_group_tag
+                    or last_role_tag != current_role_tag
+                    or bilou_utils.UNIT
+                    == bilou_utils.bilou_prefix_from_tag(current_entity_tag)
+                )
```

Comment:
Can we move this to a separate method and add some comments? Thanks.

Reply:
Or we could split the statement into multiple ones, e.g. 
```
role_or_group_changed = last_group_tag != current_group_tag or last_role_tag != current_role_tag
new_bilou_tag_begins = last_entity_tag != current_entity_tag and (bilou_utils.LAST != bilou_utils.bilou_prefix_from_tag(current_entity_tag) and bilou_utils.INSIDE != bilou_utils.bilou_prefix_from_tag(current_entity_tag) )
new_tag_starts = last_entity_tag == NO_ENTITY_TAG or bilou_utils.UNIT == bilou_utils.bilou_prefix_from_tag(current_entity_tag)

return new_tag_starts or role_or_group_changed or new_bilou_tag_begins
```
Not sure about the names. Maybe you can come up with better ones.

Reply:
Sure, made the changes. Thanks.